src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs

```diff
@@ -272,34 +272,34 @@ private static ISetupInstance LocateVisualStudioInstance(ImmutableHashSet<string
                 return isMatch;
             });
 
-            var instanceFoundWithInvalidState = false;
+            var messages = new List<string>();
 
             foreach (ISetupInstance2 instance in instances)
             {
-                var packages = instance.GetPackages()
-                                       .Where((package) => requiredPackageIds.Contains(package.GetId()));
+                var instancePackagesIds = instance.GetPackages().Select(p => p.GetId()).ToHashSet();
+                var missingPackageIds = requiredPackageIds.Where(p => !instancePackagesIds.Contains(p)).ToList();
 
-                if (packages.Count() != requiredPackageIds.Count())
+                if (missingPackageIds.Count > 0)
                 {
+                    messages.Add("An instance was found but was missing these packages: " +
```

Comment:
💭 What about including the path and ID of the instance here?

Reply:
I did path and description. I didn't include the ID because I'm not sure it's useful...